### PR TITLE
API: ArrowExtensionArray.value_counts returns pyarrow.int64 type

### LINKED ISSUE #51462: API: ArrowExtensionArray.value_counts should return int64[pyarrow] instead of Int64
Currently this explicitly calls an astype

```
return Series(counts, index=index, name="count").astype("Int64")
```

But with `ArrowExtensionArray` now able to support all pyarrow types, this should preserve the pyarrow type and return `int64[pyarrow]`

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] closes #51462 (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/algorithms.py [hunk 1]
```diff
@@ -873,6 +873,9 @@ def value_counts(
             result.name = name
             result.index.name = index_name
             counts = result._values
+            if not isinstance(counts, np.ndarray):
+                # e.g. ArrowExtensionArray
+                counts = np.asarray(counts)
```
Comment (jbrockmendel) [inline pandas/core/algorithms.py:878]: why is this needed?  id expect pd.value_counts to behave like obj.value_counts

Comment (mroeschke) [reply pandas/core/algorithms.py:878]: For `normalize` below, `counts.sum()` is called and  `._values` which is an ArrowExtensionArray doesn't have `sum`.

I think it was assumed that `counts` is a numpy array here?

Comment (mroeschke) [reply pandas/core/algorithms.py:878]: If EA's implement `value_counts` they should also handle `normalize`?

Comment (jbrockmendel) [reply pandas/core/algorithms.py:878]: > I think it was assumed that counts is a numpy array here?

For a long time I wanted value_counts to have int64 dtype and get rid of Int64 cases, but eventually gave up on that.  So i think at least Int64 is possible.  IIRC value_counts is only kinda-sorta part of the EA interface so the exact requirements aren't pinned down.

> If EA's implement value_counts they should also handle normalize?

If the normalization logic can live exclusively here, I think I'd prefer that.

> which is an ArrowExtensionArray doesn't have sum.

Out of scope, but I'd just put sum and the other most-basic reductions (say, all the ones in the np.ndarray namespace) on EA and let it raise TypeError for non-supported dtypes.

Comment (mroeschke) [reply pandas/core/algorithms.py:878]: Would you prefer me to put `sum` on ArrowExtensionArray so this works without the `if` statement or should we leave in this check?

Comment (jbrockmendel) [reply pandas/core/algorithms.py:878]: i guess cleaner to leave this and can do another api change later if called for

File: pandas/tests/extension/test_arrow.py [hunk 2]
```diff
@@ -758,19 +758,29 @@ def test_diff(self, data, periods, request):
...
-            check_stacklevel=False,
-        ):
-            super().test_value_counts_with_normalize(data)
+        data = data[:10].unique()
+        values = np.array(data[~data.isna()])
+        ser = pd.Series(data, dtype=data.dtype)
+
+        result = ser.value_counts(normalize=True).sort_index()
+
+        if not isinstance(data, pd.Categorical):
```
Comment (jbrockmendel) [inline pandas/tests/extension/test_arrow.py:774]: we can get here with Categorical?

Comment (mroeschke) [reply pandas/tests/extension/test_arrow.py]: Good point, yeah we can't. This is a copy/paste from the base class. Removed

**Comment by mroeschke** [comment]
Greenish

**Review by jbrockmendel — APPROVED** [review]
LGTM

**Comment by jbrockmendel** [comment]
thanks @mroeschke
